web3-react/src/libs/connections.ts

```diff
@@ -0,0 +1,109 @@
+import { Web3ReactHooks } from '@web3-react/core'
+import { AddEthereumChainParameter, Connector } from '@web3-react/types'
+
+import { buildCoinbaseWalletConnector } from './coinbase'
+import { CHAIN_INFO } from './constants'
+import { buildGnosisSafeConnector } from './gnosis'
+import { buildInjectedConnector } from './injected'
+import { buildNetworkConnector } from './network'
+import { buildWalletConnectConnector } from './wallet-connect'
+
+export interface Connection {
+  connector: Connector
+  hooks: Web3ReactHooks
+  type: ConnectionType
+}
+
+export enum ConnectionType {
```

Comment:
question: do we rely on the actual `enum` value? Do we need internal enum representation to be a string, matching the key? If not, I would get rid of it and make it a simple TypeScript enum where internal representation is numbers starting from `0`

Reply:
Good catch - we need this since it renders the string on the UI